pkg/agent/openflow/client.go

```diff
@@ -754,17 +751,28 @@ func (c *client) UninstallEndpointFlows(protocol binding.Protocol, endpoints []p
 	return c.deleteFlowsWithMultipleKeys(c.featureService.cachedFlows, flowCacheKeys)
 }
 
-func (c *client) InstallServiceFlows(groupID binding.GroupIDType, svcIP net.IP, svcPort uint16, protocol binding.Protocol, affinityTimeout uint16, nodeLocalExternal bool, svcType v1.ServiceType, nested bool) error {
+func (c *client) InstallServiceFlows(groupID, extraGroupID binding.GroupIDType, svcIP net.IP, svcPort uint16, protocol binding.Protocol, affinityTimeout uint16, svcType v1.ServiceType, svcOptionsFn ...config.ServiceOptionsFn) error {
 	c.replayMutex.RLock()
 	defer c.replayMutex.RUnlock()
+	svcOptions := &config.ServiceOptions{}
+	for _, fn := range svcOptionsFn {
+		fn(svcOptions)
+	}
+	externalPolicyLocal := svcOptions.ExternalPolicyLocal
+	isNestedService := svcOptions.IsNestedService
+	needShortCircuiting := svcOptions.NeedShortCircuiting
```

Comment:
Isn't NeedShortCircuiting determined by ServiceType and ExternalPolicyLocal? Why an extra parameter is needed?

Reply:
This has been removed in the latest commit.